### PR TITLE
[#55962] Add the project list view, including the "project folder" custom column

### PR DESCRIPTION
**Pull request by judithroth**
https://community.openproject.org/work_packages/55962

### CODE REVIEW

File: modules/storages/app/models/queries/storages/projects/filter/storages_filter.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      else
+        raise "unsupported operator"
+      end
+    end
+
+    def where
+      nil
+    end
+  end
+end
```
Comment (akabiru) [inline modules/storages/app/models/queries/storages/projects/filter/storages_filter.rb:68]: 🟠 Seems we're missing a unit test for the storage filter- would be worth adding one I think. Ex. https://github.com/opf/openproject/blob/dev/spec/models/queries/projects/filters/available_project_attributes_filter_spec.rb

**Review by akabiru — DISMISSED** [review]
Looking great Judith! 👍🏾 One proposal to add a missing unit spec, and should be good to go! 👍🏾

File: modules/storages/spec/features/storages/admin/project_storages_spec.rb [hunk 2]
```diff
@@ -37,11 +37,30 @@
...
+           project:,
+           storage:,
+           project_folder_mode: "automatic")
+  end
+  shared_let(:archived_project_project_storage) do
+    create(:project_storage,
+           project: archived_project,
+           storage:,
+           project_folder_mode: "inactive")
+  end
```
Comment (akabiru) [inline modules/storages/spec/features/storages/admin/project_storages_spec.rb:63]: 🟢  _nit_: I personally prefer the compact one-liner approach where possible, reduces "setup fatigue" IMO. The faster we can get to the tests without scrolling, the better 😄

File: modules/storages/spec/features/storages/admin/project_storages_spec.rb [hunk 3]
```diff
@@ -37,11 +37,30 @@
...
+           :as_automatically_managed,
+           name: "My Nextcloud Storage")
+  end
+  shared_let(:project_storage) do
+    create(:project_storage,
+           project:,
+           storage:,
+           project_folder_mode: "automatic")
+  end
+  shared_let(:archived_project_project_storage) do
```
Comment (akabiru) [inline modules/storages/spec/features/storages/admin/project_storages_spec.rb:58]: Ah, seems there was already a typo 🧹 

```suggestion
  shared_let(:archived_project_storage) do
```

File: modules/storages/app/models/queries/storages/projects/filter/storages_filter.rb [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    end
+
+    def type
+      :list
+    end
+
+    def allowed_values
+      @allowed_values ||= Storages::Storage
+                            .pluck(:name, :id)
+    end
```
Comment (akabiru) [inline modules/storages/app/models/queries/storages/projects/filter/storages_filter.rb:47]: Nice catch 👍🏾

**Review by akabiru — APPROVED** [review]
🚀

File: modules/storages/spec/features/storages/admin/project_storages_spec.rb (hunk 3 above)
Comment (akabiru) [reply modules/storages/spec/features/storages/admin/project_storages_spec.rb:58]: Ah, it's not a typo! 🤦🏾

File: modules/storages/spec/features/storages/admin/project_storages_spec.rb (hunk 2 above)
Comment (judithroth) [reply modules/storages/spec/features/storages/admin/project_storages_spec.rb:63]: I somewhat agree - it depends on how long the line gets. Sometimes it's also easier to see what the differences are between certain test objects. But if you only need 2 or 3 short arguments, I'd also go for oneliners :+1:

**Comment by judithroth** [comment]
@akabiru ah, I missed that you already approved :smile: No need to do another review, I'll merge now :slightly_smiling_face:

**Pull request merged**